presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -176,6 +177,12 @@ public ActualProperties visitApply(ApplyNode node, List<ActualProperties> inputP
         }
 
         @Override
+        public ActualProperties visitLateralJoin(LateralJoinNode node, List<ActualProperties> inputProperties)
+        {
+            return inputProperties.get(0); // lateral node input (outer query)
```

Comment:
It's fine to fail here if we don't expect to see a LateralJoinNode at this point. It's better than returning properties that are incorrect. This could cause whoever uses the properties to produce an incorrect plan, the LateralJoinNode may be removed by some other optimization, and NoSubqueryRelatedNodeChecked would fail to catch it.